modules/transactionpool/subscribe_test.go

```diff
@@ -0,0 +1,78 @@
+package transactionpool
+
+import (
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+	"testing"
+)
+
+// mockSubscriber receives transactions from the transaction pool it is
+// subscribed to, retaining them in the order they were received.
+type mockSubscriber struct {
+	txns []types.Transaction
+}
+
+// newMockSubscriber returns a blank mockSubscriber that is ready to subscribe
+// to a transaction pool.
+func newMockSubscriber() mockSubscriber {
+	return mockSubscriber{}
+}
+
+// ReceiveUpdatedUnconfirmedTransactions receives transactions from the
+// transaction pool and stores them in the order they were received.
+// This method allows *mockSubscriber to satisfy the
+// modules.TransactionPoolSubscriber interface.
+func (ms *mockSubscriber) ReceiveUpdatedUnconfirmedTransactions(txns []types.Transaction, _ modules.ConsensusChange) {
+	ms.txns = append(ms.txns, txns...)
+}
+
+// TestUnsubscribe checks that calling Unsubscribe on a mockSubscriber
+// shortens the list of subscribers to the transaction pool by 1 (doesn't
+// actually check that the mockSubscriber was the one unsubscribed).
+func TestUnsubscribe(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+
+	tpt, err := createTpoolTester("TestUnsubscribe")
+	if err != nil {
+		t.Fatal(err)
+	}
+	// TODO: add tPoolTester.Close() method
+	//	defer tpt.Close()
+
+	// Check the transaction pool is empty when initialized.
+	if len(tpt.tpool.transactionSets) != 0 {
+		t.Fatal("transaction pool is not empty")
+	}
+
+	// Create a mock subscriber and subscribe it to the transaction pool.
+	ms := newMockSubscriber()
+	tpt.tpool.TransactionPoolSubscribe(&ms)
+	if len(ms.txns) != 0 {
+		t.Fatalf("mock subscriber has received %v transactions; shouldn't have received any yet", len(ms.txns))
+	}
+
+	// Create a valid transaction set and check that the mock subscriber's
+	// transaction list is updated.
```

Comment:
seems like this tests more than just `Unsubscribe`, which is fine -- just change the name to `TestSubscription` (or similar) and update the docstring.